src/libraries/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -426,16 +453,32 @@ public void PathWithReservedDeviceNameAsExtendedPath(string path)
         [Theory,
             MemberData(nameof(UncPathsWithoutShareName))]
         [PlatformSpecific(TestPlatforms.Windows)]
-        public void UncPathWithoutShareNameAsPath_ThrowsIOException_Core(string path)
+        public void UncPathWithoutShareNameAsPath_ThrowsIOException(string path)
         {
-            Assert.ThrowsAny<IOException>(() => Create(path));
+            // Same exception message, different exception type
+            if (PlatformDetection.IsNetCore)
+            {
+                Assert.ThrowsAny<IOException>(() => Create(path));
+            }
+            else
+            {
+                Assert.Throws<ArgumentException>(() => Create(path));
+            }
```

Comment:
Why is this one written this way, and the test change a page above is using the Throws that takes two generic arguments?

Reply:
Because of the `ThrowsAny`. The unit tests that use `ThrowsAny` can throw any exception that inherits from `IOException`.

There are some test classes in `FileSystem` that inherit from others, and the APIs are tested in a very similar way. The only thing that changes is the type of exception expected (for example `DirectoryNotFoundException` vs `FileNotFoundException`).

It took me some time to figure out why when I tried to use `Throws<IOException>` was not working for all cases. It was because of the test class inheritance.

Would it help if I add a comment above the test class to explain this?